### ISSUE TITLE
Yield From values are not recognised as returned values in Squiz FunctionComment sniff

### ISSUE THREAD

**Issue by tomasfejfar** [squizlabs/PHP_CodeSniffer#1364]
```php
function fetchEntity(): MyEntity
{
// get it from DB
}

/**
 * @return MyEntity[]|Generator
*/
function generator(): Generator
{
    yield $this->fetchEntity();
}
```
I want to typehint the generator contents to my IDE. It's a generator, but it acts like an array of entities as well.

Unfortunatelly I end up with phpcs error:
`Function has no return statement, but annotation @return is present (Squiz.Commenting.FunctionComment.InvalidNoReturn)`

Is this expected behaviour?

**Comment by gsherwood** [comment]
This was reported as bug #479 and released in version 2.3.1 nearly 2 years ago. I've given it another test and I don't receive the error, and the unit tests are still passing, so I can't see a regression.

Are you using a really old version of PHPCS?

**Comment by tomasfejfar** [comment]
Oh, sorry I misinterpreted the problem when creating the issue. The root cause seem to be the usage of `yield from`. 

```
$ vendor\bin\phpcs --version
PHP_CodeSniffer version 2.8.1 (stable) by Squiz (http://www.squiz.net)
$ vendor\bin\phpcs --standard=ruleset.xml --extensions=php --encoding=utf-8 --tab-width=4 -sp app
...
 77 | ERROR | [ ] Function has no return statement, but annotation
    |       |     @return is present
    |       |     (Squiz.Commenting.FunctionComment.InvalidNoReturn)
```

The offending code is

```
	/**
	 * @return \App\Entity[]|\Generator  <------------ line 77
	 */
	public function createAllEntitiesGenerator(): Generator
	{
		$query = $this
			->createAllEntitiesQueryBuilder()
			->getQuery();
		yield from Util::createGeneratorFromQueryWithEmClear($query, $this->entityManager);
	}
```

**Comment by gsherwood** [comment]
Thanks for that extra code. It made the problem is to find and fix.

**Comment by tomasfejfar** [comment]
I was just trying to understand how the tests work and to send PR 🙂 It's great that you managed to find and fix it so quickly. Thanks.
